gemini/evaluation/evaluate_multimodal_task_image.ipynb

```diff
@@ -0,0 +1,758 @@
+    "    \"lateral\",\n",
+    "    \"windshield\",\n",
+    "]\n",
+    "\n",
+    "reference = [\"bumper\", \"engine_compartment\", \"hood\", \"lateral\", \"windshield\"]\n",
+    "\n",
+    "eval_dataset = pd.DataFrame(\n",
+    "    {\n",
+    "        \"instruction\": instruction,\n",
+    "        \"context\": context,\n",
+    "        \"response\": generated_response,\n",
+    "        \"reference\": reference,\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "icfOEQopmpky"
+   },
+   "outputs": [],
+   "source": [
+    "eval_dataset.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "EFUFtKRIUpAe"
+   },
+   "source": [
+    "### Bring your own autorater\n",
+    "\n",
+    "Gen AI Evaluation Service allows you to bring any autorater you prefer. In this case, you use `gemini-1.5-pro` with a typical structured output to evaluate your task."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "oV6ZI1E5Uu1c"
+   },
+   "outputs": [],
+   "source": [
+    "def get_autorater_response(metric_prompt: list) -> dict:\n",
+    "    metric_response_schema = {\n",
+    "        \"type\": \"OBJECT\",\n",
+    "        \"properties\": {\n",
+    "            \"score\": {\"type\": \"NUMBER\"},\n",
+    "            \"explanation\": {\"type\": \"STRING\"},\n",
+    "        },\n",
+    "        \"required\": [\"score\", \"explanation\"],\n",
+    "    }\n",
+    "\n",
+    "    autorater = GenerativeModel(\n",
+    "        \"gemini-1.5-pro\",\n",
+    "        generation_config=GenerationConfig(\n",
+    "            response_mime_type=\"application/json\",\n",
+    "            response_schema=metric_response_schema,\n",
+    "        ),\n",
+    "        safety_settings={\n",
+    "            HarmCategory.HARM_CATEGORY_UNSPECIFIED: HarmBlockThreshold.BLOCK_NONE,\n",
+    "            HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_NONE,\n",
+    "            HarmCategory.HARM_CATEGORY_HATE_SPEECH: HarmBlockThreshold.BLOCK_NONE,\n",
+    "            HarmCategory.HARM_CATEGORY_HARASSMENT: HarmBlockThreshold.BLOCK_NONE,\n",
+    "            HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: HarmBlockThreshold.BLOCK_NONE,\n",
+    "        },\n",
+    "    )\n",
+    "\n",
+    "    response = autorater.generate_content(metric_prompt)\n",
+    "\n",
+    "    response_json = {}\n",
+    "\n",
+    "    if response.candidates and len(response.candidates) > 0:\n",
+    "        candidate = response.candidates[0]\n",
+    "        if (\n",
+    "            candidate.content\n",
+    "            and candidate.content.parts\n",
+    "            and len(candidate.content.parts) > 0\n",
+    "        ):\n",
+    "            part = candidate.content.parts[0]\n",
+    "            if part.text:\n",
+    "                response_json = json.loads(part.text)\n",
```

Comment:
I don't think you need to do this. Should be able to just access `response_json = response.text`

Reply:
I jsonize because it is easier to handle the result in the custom metric.